src/shell/shell_misc.cpp

```diff
@@ -501,234 +443,153 @@ bool DOS_Shell::Execute(char * name,char * args) {
-		 * Qbix: It is extremly messy, but this was the only way I could get things like /:aa and :/aa to work correctly */
-		
-		//Prepare string first
-		char parseline[258] = { 0 };
-		for(char *pl = line,*q = parseline; *pl ;pl++,q++) {
-			if (*pl == '=' || *pl == ';' || *pl ==',' || *pl == '\t' || *pl == ' ') 
-				reset_str(q);
-			else
-				*q = *pl; //Replace command seperators with 0.
-		} //No end of string \0 needed as parseline is larger than line
-
-		for(char* p = parseline; (p-parseline) < 250 ;p++) { //Stay relaxed within boundaries as we have plenty of room
-			if (*p == '/') { //Transform /Hello into H\0ello
-				reset_str(p);
-				p++;
-				while ( *p == 0 && (p-parseline) < 250) p++; //Skip empty fields
-				if ((p-parseline) < 250) { //Found something. Lets get the first letter and break it up
-					p++;
-					memmove(static_cast<void*>(p + 1),static_cast<void*>(p),(250-(p-parseline)));
-					if ((p - parseline) < 250)
-						reset_str(p);
-				}
-			}
 		}
-		// Just to be safe
-		terminate_str_at(parseline, 255);
-		terminate_str_at(parseline, 256);
-		terminate_str_at(parseline, 257);
-
-		/* Parse FCB (first two parameters) and put them into the current DOS_PSP */
-		uint8_t add;
-		uint16_t skip = 0;
-		//find first argument, we end up at parseline[256] if there is only one argument (similar for the second), which exists and is 0.
-		while(skip < 256 && parseline[skip] == 0) skip++;
-		FCB_Parsename(dos.psp(),0x5C,0x01,parseline + skip,&add);
-		skip += add;
-		
-		//Move to next argument if it exists
-		while(parseline[skip] != 0) skip++;  //This is safe as there is always a 0 in parseline at the end.
-		while(skip < 256 && parseline[skip] == 0) skip++; //Which is higher than 256
-		FCB_Parsename(dos.psp(),0x6C,0x01,parseline + skip,&add); 
-
-		block.exec.fcb1=RealMake(dos.psp(),0x5C);
-		block.exec.fcb2=RealMake(dos.psp(),0x6C);
-		/* Set the command line in the block and save it */
-		block.exec.cmdtail=RealMakeSeg(ss,reg_sp+0x100);
-		block.SaveData();
-#if 0
-		/* Save CS:IP to some point where i can return them from */
-		uint32_t oldeip=reg_eip;
-		uint16_t oldcs=SegValue(cs);
-		RealPt newcsip=CALLBACK_RealPointer(call_shellstop);
-		SegSet16(cs,RealSeg(newcsip));
-		reg_ip=RealOffset(newcsip);
-#endif
-		/* Start up a dos execute interrupt */
-		reg_ax=0x4b00;
-		//Filename pointer
-		SegSet16(ds,SegValue(ss));
-		reg_dx=RealOffset(file_name);
-		//Paramblock
-		SegSet16(es,SegValue(ss));
-		reg_bx=reg_sp;
-		SETFLAGBIT(IF,false);
-		CALLBACK_RunRealInt(0x21);
-		/* Restore CS:IP and the stack */
-		reg_sp+=0x200;
-#if 0
-		reg_eip=oldeip;
-		SegSet16(cs,oldcs);
-#endif
+		bf   = std::make_shared<BatchFile>(this,
+                                                 fullname.c_str(),
+                                                 std::string(name).c_str(),
+                                                 std::string(args).c_str());
+		echo = temp_echo;
```

Comment:
(indented whitespace needs reformatting)

Maybe can use `name.data()` and `args.data()` directly from the string_views?

Reply:
> using c_str() is more correct, since data() doesn't guarantee null termination, which the called function needs.

roger that :+1: